docs/core_docs/docs/modules/experimental/mask/mask.mdx

```diff
@@ -0,0 +1,64 @@
+# Masking
+
+The experimental masking parser and transformer is an extendable module for masking and rehydrating strings. One of the primary use cases for this module is to redact PII (Personal Identifiable Information) from a string before making a call to an llm.
+
+### Real world scenario
+
+A customer support system receives messages containing sensitive customer information. The system must parse these messages, mask any PII (like names, email addresses, and phone numbers), and log them for analysis while complying with privacy regulations. Before logging the transcript a summary is generated using an llm.
+
+## Example
+
+Use the RegexMaskingTransformer to create a simple mask for email and phone.
+
+```typescript
+import {
+  MaskingParser,
+  RegexMaskingTransformer,
+} from "langchain/experimental/masking";
+
+// Define masking strategy
+const emailMask = () => `[email-${Math.random().toString(16).slice(2)}]`;
+const phoneMask = () => `[phone-${Math.random().toString(16).slice(2)}]`;
+
+// Configure pii transformer
+const piiMaskingTransformer = new RegexMaskingTransformer({
+  email: { regex: /\S+@\S+\.\S+/g, mask: emailMask },
+  phone: { regex: /\d{3}-\d{3}-\d{4}/g, mask: phoneMask },
+});
+
+maskingParser = new MaskingParser();
+maskingParser.addTransformer(piiMaskingTransformer);
```

Comment:
Would it make more sense for the constructor to just take an array of `MaskingTransformers`?

Reply:
I will update the basic example to default to an array of transformers.

```typescript
const piiMaskingTransformer = new RegexMaskingTransformer({
  email: { regex: /\S+@\S+\.\S+/g, mask: emailMask },
  phone: { regex: /\d{3}-\d{3}-\d{4}/g, mask: phoneMask },
});

const maskingParser = new MaskingParser({transformers: [piiMaskingTransformer]});